Signal/src/AppDelegate.m

```diff
@@ -1039,6 +1057,28 @@ - (void)application:(UIApplication *)application
     }];
 }
 
+- (void)application:(UIApplication *)application
+    performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult result))completionHandler
+{
+    DDLogInfo(@"%@ performing background fetch", self.logTag);
+    [AppReadiness runNowOrWhenAppIsReady:^{
+        __block AnyPromise *job = [[SignalApp sharedApp].messageFetcherJob run].then(^{
+            // HACK: Call completion handler after n seconds.
+            //
+            // We don't currently have a convenient API to know when message fetching is *done* when
+            // working with the websocket.
+            //
+            // We *could* substantially rewrite the TSSocketManager to take advantage of the `empty` message
+            // But once our REST endpoint is fixed to properly de-enqueue fallback notifications, we can easily
+            // use the rest endpoint here rather than the websocket and circumvent making changes to critical code.
+            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
```

Comment:
So my reading of this comment is that five is simply a very conservative estimate for "log enough to receive and ack but not necessarily process at least 1 message/one empty queue message."

Reply:
Correct